### ISSUE TITLE
Fatal runtime error CLI OSX

### ISSUE THREAD

**Issue by rpmirish** [cloudfoundry/cli#783]
intstalled cf CLI, when attempt to run the following throws:

fatal error: runtime: bsdthread_register error

runtime stack:
runtime.throw(0x9e5a20, 0x21)
	/usr/local/go/src/runtime/panic.go:530 +0x90 fp=0x7fff5fbffb10 sp=0x7fff5fbffaf8
runtime.goenvs()
	/usr/local/go/src/runtime/os1_darwin.go:76 +0x8d fp=0x7fff5fbffb38 sp=0x7fff5fbffb10
runtime.schedinit()
	/usr/local/go/src/runtime/proc.go:440 +0xa7 fp=0x7fff5fbffb80 sp=0x7fff5fbffb38
runtime.rt0_go(0x7fff5fbffbb8, 0x1, 0x7fff5fbffbb8, 0x0, 0x0, 0x1, 0x7fff5fbffcb8, 0x0, 0x7fff5fbffcf3, 0x7fff5fbffd0f, ...)
	/usr/local/go/src/runtime/asm_amd64.s:138 +0x132 fp=0x7fff5fbffb88 sp=0x7fff5fbffb80

can't seem to get it working. Running Mac with Yosemite 10.10.5

**Comment by dkoper** [comment]
Hi @rpmirish,

"when attempt to run the following"... Can you provide more information, such as the commands you tried, cf CLI release version, etc. (see https://github.com/cloudfoundry/cli#filing-bugs)?

Cheers,
Dies Koper
CF CLI PM

**Comment by kchawla19** [comment]
Same issue, just typing "cf" after install gives the above error on yosemite 10.0.5. you should be able to reproduce very easily

**Comment by cbguder** [comment]
@rpmirish @kchawla19 we believe this error is happening due to Anti-Virus/Security software. do you have one of the following, or other Anti-Virus/Security software installed?

* McAfee Endpoint Protection for Mac
* Trend Micro Data Loss Prevention
* Websense Endpoint
* Digital Guardian

**Comment by dkoper** [comment]
@rpmirish  @kchawla19 Also, if you temporary disable your antvirus/security software, does cf 6.16.x work?
Knowing that would help us tremendously in pinpointing the cause of the issue.

**Comment by shinji62** [comment]
HI, 6.16.X do not work for me neither on mac 10.10.2 Yosemite 
```
Error performing request: Get https://api.run.pivotal.io/v2/info: dial tcp: i/o timeout
TIP: If you are behind a firewall and require an HTTP proxy, verify the https_proxy environment variable is correctly set. Else, check your network connection.
```

I have the McAfee Endpoint Protection for Mac  (2.2.0) but difficult to desactivate (Corporate Laptop)

CLI 6.15.x works well

**Comment by benschenker** [comment]
The command: `export DYLD_INSERT_LIBRARIES=` seems to fix this error for me... See https://github.com/direnv/direnv/issues/194

**Comment by cbguder** [comment]
@benschenker when you get an error, does it say:

`runtime: bsdthread_register error`

or

`runtime: bsdthread_register error (unset DYLD_INSERT_LIBRARIES)`

**Comment by benschenker** [comment]
the former
```
fatal error: runtime: bsdthread_register error

runtime stack:
runtime.throw(0x9e5a20, 0x21)
	/usr/local/go/src/runtime/panic.go:530 +0x90 fp=0x7fff5fbff0e0 sp=0x7fff5fbff0c8
runtime.goenvs()
	/usr/local/go/src/runtime/os1_darwin.go:76 +0x8d fp=0x7fff5fbff108 sp=0x7fff5fbff0e0
runtime.schedinit()
	/usr/local/go/src/runtime/proc.go:440 +0xa7 fp=0x7fff5fbff150 sp=0x7fff5fbff108
runtime.rt0_go(0x7fff5fbff180, 0x1, 0x7fff5fbff180, 0x0, 0x1, 0x7fff5fbff400, 0x0, 0x7fff5fbff43b, 0x7fff5fbff47d, 0x7fff5fbff4c9, ...)
	/usr/local/go/src/runtime/asm_amd64.s:138 +0x132 fp=0x7fff5fbff158 sp=0x7fff5fbff150
```

**Comment by cbguder** [comment]
@rpmirish @kchawla19 @shinji62 @benschenker

We have built a new binary that we think might fix this issue. Please give it a try and let us know if it fixes the issue you're seeing.

Installer: https://s3.amazonaws.com/go-cli/releases/v6.16.1-2016-03-17/cf-cli-installer_6.16.1_osx.pkg

Binary: https://s3.amazonaws.com/go-cli/releases/v6.16.1-2016-03-17/cf-cli_6.16.1_osx.tgz

**Comment by dkoper** [comment]
Hi @rpmirish, @kchawla19, @shinji62, @benschenker,

We have ported the fix into our edge binaries. Could try out our edge binary and confirm the issue is addressed?
https://cli.run.pivotal.io/edge?arch=macosx64&source=github

Cheers,
Dies Koper
CF CLI PM

**Comment by shinji62** [comment]
Seems ok for me.

**Comment by dkoper** [comment]
Thanks @shinji62!

**Comment by Devyno** [comment]
Had the same issue earlier and the new binary fixed the problem.. Thanks

**Comment by dkoper** [comment]
The fix has been included in cf CLI v6.17.0.
https://github.com/cloudfoundry/cli#downloads

Thanks again!
Dies Koper
CF CLI PM
